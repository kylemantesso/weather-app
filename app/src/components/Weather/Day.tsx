import React from "react";
import moment from "moment";

import { Box, Text } from "grommet";
import { Icon } from "./Icon";

export const Day: React.FC<{
  icon: string;
  description: string;
  day: string;
  today: boolean;
  max: number;
  min: number;
}> = ({ icon, day, today, max, min, description }) => (
  <Box direction="column" justify="center" align="center" gap="small">
    <Text weight="bold" size="small">
      {today ? "Today" : moment(day).format("dddd")}
    </Text>
    <Icon icon={icon} description={description} />
    <Box direction="row" gap="xsmall">
      <Text weight="bold" size="small">
        {Math.round(max)}°
      </Text>
      <Text size="small">{Math.round(min)}°</Text>
    </Box>
  </Box>
);
