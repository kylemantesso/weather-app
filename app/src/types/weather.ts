export interface ILocationResult {
  title: string;
  location_type: string;
  woeid: number;
  latt_long: string;
}

export interface IConsolidatedWeather {
  id: any;
  weather_state_name: string;
  weather_state_abbr: string;
  wind_direction_compass: string;
  created: Date;
  applicable_date: string;
  min_temp: number;
  max_temp: number;
  the_temp: number;
  wind_speed: number;
  wind_direction: number;
  air_pressure: number;
  humidity: number;
  visibility: number;
  predictability: number;
}

interface IParent {
  title: string;
  location_type: string;
  woeid: number;
  latt_long: string;
}

interface ISource {
  title: string;
  slug: string;
  url: string;
  crawl_rate: number;
}

export interface IWeatherResult {
  consolidated_weather: IConsolidatedWeather[];
  time: Date;
  sun_rise: Date;
  sun_set: Date;
  timezone_name: string;
  parent: IParent;
  sources: ISource[];
  title: string;
  location_type: string;
  woeid: number;
  latt_long: string;
  timezone: string;
}

export interface IWeather {
  title: string;
  current: IDay;
  forecast: IDay[];
}

export interface IDay {
  id: string;
  date: string;
  description: string;
  temp?: number;
  icon: string;
  windSpeed: string;
  windDirection: string;
  humidity: number;
  max: number;
  min: number;
}
