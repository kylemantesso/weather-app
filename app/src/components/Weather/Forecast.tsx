import React, { useContext } from "react";
import { Box, Grid, ResponsiveContext } from "grommet";

import { Day } from "./Day";
import { IDay } from "../../types/weather";

export const Forecast: React.FC<{ forecasts: IDay[] }> = ({ forecasts }) => (
  <Box pad={{ left: "medium", right: "medium", top: "medium" }}>
    <Grid columns="xsmall" align="center">
      {forecasts.map((forecast, ix) => (
        <Day
          description={forecast.description}
          min={forecast.min}
          max={forecast.max}
          key={forecast.id}
          today={ix === 0}
          day={forecast.date}
          icon={forecast.icon}
        />
      ))}
    </Grid>
  </Box>
);
