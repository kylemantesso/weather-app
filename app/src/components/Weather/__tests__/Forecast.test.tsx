import React from "react";
import renderer from "react-test-renderer";
import { Day } from "../Day";
import { IDay } from "../../../types/weather";
import { Forecast } from "../Forecast";

describe("Forecast", () => {
  it("matches snapshot", () => {
    const forecasts: IDay[] = [
      {
        id: "a",
        date: "2020-08-01",
        description: "Light rain",
        temp: 20.5,
        icon: "lr",
        windSpeed: "10 kph",
        windDirection: "S",
        humidity: 80,
        max: 34,
        min: 12,
      },
      {
        id: "b",
        date: "2020-08-02",
        description: "Clear",
        icon: "c",
        windSpeed: "20 kph",
        windDirection: "S",
        humidity: 80,
        max: 30,
        min: 15,
      },
    ];
    const component = renderer.create(<Forecast forecasts={forecasts} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
