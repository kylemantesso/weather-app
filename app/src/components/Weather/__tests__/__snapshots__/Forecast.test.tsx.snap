// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast matches snapshot 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 cxtNjs"
>
  <div
    className="StyledGrid-sc-1wofa1l-0 lgHuga"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 fHyGHh"
    >
      <span
        className="StyledText-sc-1sadyjn-0 bDBJwM"
        size="small"
      >
        Today
      </span>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
      />
      <img
        alt="Light rain"
        height="60px"
        src="https://www.metaweather.com/static/img/weather/lr.svg"
      />
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
      />
      <div
        className="StyledBox-sc-13pk1d4-0 bIdaiR"
      >
        <span
          className="StyledText-sc-1sadyjn-0 bDBJwM"
          size="small"
        >
          34
          째
        </span>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 cPZcyD"
        />
        <span
          className="StyledText-sc-1sadyjn-0 hncrgS"
          size="small"
        >
          12
          째
        </span>
      </div>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fHyGHh"
    >
      <span
        className="StyledText-sc-1sadyjn-0 bDBJwM"
        size="small"
      >
        Sunday
      </span>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
      />
      <img
        alt="Clear"
        height="60px"
        src="https://www.metaweather.com/static/img/weather/c.svg"
      />
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
      />
      <div
        className="StyledBox-sc-13pk1d4-0 bIdaiR"
      >
        <span
          className="StyledText-sc-1sadyjn-0 bDBJwM"
          size="small"
        >
          30
          째
        </span>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 cPZcyD"
        />
        <span
          className="StyledText-sc-1sadyjn-0 hncrgS"
          size="small"
        >
          15
          째
        </span>
      </div>
    </div>
  </div>
</div>
`;
